File: 2ºCurso/PrimerCuatri/PDOO/Practicas/P1/Ejercicios/PruebasJava/src/Persona.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pruebasjava;

/**
 *
 * @author ylope
 */
public class Persona {
    // Atributo y Método de clase
    static private int numPersonas = 0;
    static int getNumPersonas() {
        return numPersonas;
    }
    //Atributo de instancia 
    private String nombre;
    //Incializador
    Persona ( String unNombre) {
        nombre = unNombre;
        numPersonas++;
    }
    
}

File: 2ºCurso/PrimerCuatri/PDOO/Practicas/P1/Ejercicios/PruebasRuby/Persona.rb
class Persona
  #atributo y metodo de clase
  @@num_personas = 0
  def self.num_personas
    @@num_personas
  end
  #inicializador
  def initialize (un_nombre)
    #atributo de instancia
    @nombre = un_nombre
    @@num_personas += 1
  end

  def get_numPersonas()
    return @@num_personas
  end
end